centos_ci.sh

```diff
@@ -322,6 +328,12 @@ function build_and_test() {
   # Run integration test with 'kvm' driver
   MINISHIFT_VM_DRIVER=kvm make integration_pr
   echo "CICO: Tests ran successfully"
+
+  # Publish documentation for master build only
+  if [[ "$GIT_BRANCH" = "origin/master" ]]; then
```

Comment:
What does this actually mean here? Are we now building and hosting the docs as part of each pull request? Or is this just for the push to master. If the latter, why do we handle the master push the same way as the pull request build? Really, the distinction between master and pull request build should be made at the end of the script where we switch on the job name. In the last else branch we should distinguish between master and pull request and call functions accordingly. 
In this case we don't have to have this sneaky not quite clear if statement. 

More importantly, if we have a dedicated method for the master build we can also let the full test suite run, instead of just the pull request integration test. We are giving an opportunity of our hands to get feedback early. 

Bottom line:
* The job switch statement should already determine master vs pull request build and call dedicated functions
* The function of the master build should include the docs build and publish
* On the master build we could also run all integration tests (does the nightly pass?)

Reply:
> * The job switch statement should already determine master vs pull request build and call dedicated functions
> * The function of the master build should include the docs build and publish
> * On the master build we could also run all integration tests (does the nightly pass?)

@hferentschik  All addressed in recent refactoring.

Yes, nightly build is passing. See the last passed nightly build https://ci.centos.org/job/minishift-nightly/19/consoleFull